lib/faktory/connection.ex

```diff
@@ -104,16 +104,36 @@ defmodule Faktory.Connection do
     end
   end
 
-  defp handshake!(socket, wid) do
+  defp handshake!(socket, wid, password) do
     :inet.setopts(socket, packet: :line)
-    {:ok, <<"+HI", _rest::binary>>} = :gen_tcp.recv(socket, 0)
+    {:ok, <<"+HI", rest::binary>>} = :gen_tcp.recv(socket, 0)
+
+    server_config = Poison.decode!(rest)
+    salt = server_config["s"]
+    server_version = to_int(server_config["v"])
+
+    if server_version > 2 do
+      Logger.warn("Warning: Faktory server protocol #{server_version} in use, but this worker doesn't speak that version")
+    end
+
+    password_opts =
+      if salt = server_config["s"] do
```

Comment:
Should this be `=` or `==` here?

Reply:
Actually I think I understand it now. May be extracting this to a separate function will make it more readable?

Reply:
:+1: I made the change.

The `if foo = could_return_nil() ... else ...` syntax isn't well-documented, but it's useful sometimes. I agree that pattern matching is better in this case, though.